sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -193,52 +195,45 @@ static const dif_alert_handler_class_config_t
             },
 };
 
-static const alert_info_t kExpectedInfo[kRoundTotal] = {
+static alert_info_t kExpectedInfo[kRoundTotal] = {
     [kRound1] =
         {
             .test_name = "Single class(ClassA)",
-            .alert_cause = ((uint64_t)1 << kTopEarlgreyAlertIdI2c0FatalFault) |
-                           ((uint64_t)1 << kTopEarlgreyAlertIdI2c1FatalFault) |
-                           ((uint64_t)1 << kTopEarlgreyAlertIdI2c2FatalFault),
             .class_accum_cnt = {3, 0, 0, 0},
             .class_esc_state = {kCstatePhase0, kCstateIdle, kCstateIdle,
                                 kCstateIdle},
         },
     [kRound2] =
         {
             .test_name = "Multi classes(ClassB,C)",
-            .alert_cause =
-                ((uint64_t)1 << kTopEarlgreyAlertIdUart0FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdUart1FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdUart2FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdUart3FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdOtpCtrlFatalBusIntegError),
             .class_accum_cnt = {0, 1, 4, 0},
             .class_esc_state = {kCstateIdle, kCstatePhase1, kCstatePhase0,
                                 kCstateIdle},
         },
     [kRound3] =
         {
             .test_name = "All classes",
-            .alert_cause =
-                ((uint64_t)1 << kTopEarlgreyAlertIdRvCoreIbexRecovSwErr) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdUart0FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdI2c0FatalFault) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdSpiHost0FatalFault),
             .class_accum_cnt = {1, 1, 1, 1},
             .class_esc_state = {kCstatePhase0, kCstatePhase1, kCstatePhase0,
                                 kCstatePhase0},
         },
 };
 
+// TODO: This entire function needs to be made smarter.
+// There is too much hardcoding that changes when we add / subtract
+// alerts.
+// We need to track both a dump_idx and a bit_idx to know where we
+// are and do the conversion in an automatic way.
```

Comment:
Agreed. Challenge I had was minimum resolution of c struct is 8bit while verilog struct defines 1 bit resolution.

Reply:
i feel like there might be some clever things we can do with C struct bit fields and unions...but honestly would need to look into that to be sure.